### ISSUE TITLE
salt 2019: regression in mount persistance

### ISSUE THREAD

**Issue by disaster123** [saltstack/salt#54084]
Since upgrading to 2019.2 and to py3 the mount persistance updates fstab on every state.apply. This is because the mount options are in no order and get randomly inserted into fstab.

Ex:
```
# grep cache /etc/fstab 
tmpfs		/home/cache	tmpfs	size=4G,mode=1777,defaults,nofail	0 0

salt-call state.apply
...
          ID: freitext-name mount
    Function: mount.mounted
        Name: /home/cache
      Result: True
     Comment: Target was already mounted. Updated the entry in the fstab.
     Started: 08:17:55.450982
    Duration: 48.521 ms
     Changes:   
              ----------
              persist:
                  update
...

# grep cache /etc/fstab 
tmpfs		/home/cache	tmpfs	defaults,nofail,size=4G,mode=1777	0 0

```

**Comment by disaster123** [comment]
This could be easily fixed by sorting opts:
i added the following lines to mount.py just before the for opt in opts loop starts:
```
392                 if opts:
393                     opts.sort()
```

**Comment by garethgreenaway** [comment]
@disaster123 Thanks for the report.  So far I haven't been able to reproduce this.  Can you provide the state file that you're using to see this issue?  Thanks!

**Comment by disaster123** [comment]
Be sure you use python3.

Example state:
```
mount /home/tmp:
  mount.mounted:
    - name: /home/tmp
    - device: /home/tmp
    - fstype: none
    - opts:
      - bind
      - nodev
      - noexec
      - nosuid
      - rw
    - hidden_opts:
      - nodev
      - noexec
      - nosuid
    - persist: True
    - require:
      - file: create /home/tmp
```

**Comment by garethgreenaway** [comment]
@disaster123 Thanks!  I am able to reproduce the issue now.

**Comment by disaster123** [comment]
Can you than please remove / fix the labels?

**Comment by disaster123** [comment]
Why is this still unfixed in 2ß19.2.2?

**Comment by disaster123** [comment]
no this issue is still valid

**Comment by disaster123** [comment]
@sagetherage this one is also still unfixed

**Comment by sagetherage** [comment]
unfixed in 2019.2.x?  I see there is a fix in master: I will follow up with @garethgreenaway to get more information

**Comment by disaster123** [comment]
Could you point me to the commit? And is it in 2019.2?

**Comment by sagetherage** [comment]
no, the commit is in master -- > let me get with Gareth, busy today, but still working on this.

**Comment by sagetherage** [comment]
@disaster123 this brought up some discussion points that aren't necessarily yours or to be made here, but wanted to update you I am having conversations and still trying to find out about fixing it in an older branch. It is fixed in master and merged and will be released with Neon. More to come about a fix in 2019.2 I apologize this is taking longer than I expected.

**Comment by garethgreenaway** [comment]
This issue is fixed in the latest release and with our new branching / release strategy we won't be back porting changes into older releases unless the change is related to a security fix.  If the issue persists in 3000.1 and future release please open a new issue, closing this one out.
